### PR TITLE
config,store/tikv: make how large a txn start ttlManager configurable

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

If a transaction need to update its lock TTL, it start a ttlManager. Usually this is decided by the transaction size.

If the txn size is small and we start a ttlManager, it would bring unnecessary performance penalty.
If the txn size is large and do not start a ttlManager, the txn lock would gone while the txn is active, cause commit action fail.

This PR make the size configurable.

### What is changed and how it works?

What's Changed:

Introduce a config `ttl-based-txn-size`.

How it Works:

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

-  Add a config item `ttl-based-txn-size` to make how large a transaction should update the lock TTL configurable.

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @lysu @jackysp

File: config/config.toml.example [hunk 1]
```diff
@@ -369,6 +369,10 @@ store-limit = 0
 # store-liveness-timeout is used to control timeout for store liveness after sending request failed.
 store-liveness-timeout = "5s"
 
+# ttl-based-txn-size decides whether a transaction should update its lock TTL.
+# If the size(in byte) of a transaction is large than `ttl-based-txn-size`, it update the lock TTL during the 2PC.
```
Comment (jackysp) [inline config/config.toml.example:373]: should we mention that it only affects the optimistic transaction?

Comment (tiancaiamao) [reply config/config.toml.example:373]: It also affect the pessimistic transaction if the pessimistic does not write locks before 2PC.

**Pull request merged**